contracts/Content/UniqueContent.sol

```diff
@@ -93,22 +91,19 @@ contract UniqueContent is IUniqueContent, IMultipleRoyalties, MultipleRoyalties,
     */
     function tokenURI(uint256 _uniqueId) public view override returns (string memory) {
         require(_exists(_uniqueId), "Unique Id does not exist");
-        return uniqueAssetInfo[_uniqueId].uniqueAssetUri[uniqueAssetInfo[_uniqueId].version];
+        return uniqueContentStorage.tokenURI(_uniqueId, type(uint256).max);
     }
 
     /**
     * @dev If the caller is creator and owner of the token, it adds a new version of the unique asset
-    * @param _uniqueId uint256 ID of the token to set its uri
+    * @param _uniqueId uint256 ID of the token that gets a new uri
     * @param _uri string URI to assign
     */
     function setUniqueUri(uint256 _uniqueId, string memory _uri) external override {
         require(_exists(_uniqueId), "Unique Id does not exist");
-        require(uniqueAssetInfo[_uniqueId].creator == _msgSender(), "You are not the creator of this token");
         require(ownerOf(_uniqueId) == _msgSender(), "You must be the owner of this token to set its uri");
-        uniqueAssetInfo[_uniqueId].uniqueAssetUri.push(_uri);
-        uniqueAssetInfo[_uniqueId].version++;
-
-        emit UniqueUriUpdated(_uniqueId, uniqueAssetInfo[_uniqueId].version);
+        require(uniqueContentStorage.isCreator(_uniqueId, _msgSender()), "You are not the creator of this token");
```

Comment:
Actually, long error messages like this contributes to the size of the contract. You could reduce this to something like "Error: Sender not Token Creator" or "Error: Sender not token owner"

Reply:
Same for other error messages. try to keep them short and brief